shared/teams/common/channels-widget.tsx

```diff
@@ -0,0 +1,142 @@
+import * as React from 'react'
+import * as Kb from '../../common-adapters'
+import * as Styles from '../../styles'
+import * as Container from '../../util/container'
+import * as Constants from '../../constants/teams'
+import * as Types from '../../constants/types/teams'
+import ChannelPopup from '../team/settings-tab/channel-popup'
+import useAutocompleter from './use-autocompleter'
+
+type Props = {
+  channels: Array<string>
+  onAddChannel: (channelname: string) => void
+  onRemoveChannel: (channelname: string) => void
+  teamID: Types.TeamID
+}
+
+// always shows #general
+const ChannelsWidget = (props: Props) => {
+  return (
+    <Kb.Box2 direction="vertical" gap="tiny" style={styles.container}>
+      <ChannelInput onAdd={props.onAddChannel} teamID={props.teamID} selected={props.channels} />
+      <Kb.Box2 direction="horizontal" gap="xtiny" fullWidth={true} style={styles.pillContainer}>
+        <ChannelPill channelname="general" />
+        {props.channels.map(channelname => (
+          <ChannelPill
+            key={channelname}
+            channelname={channelname}
+            onRemove={() => props.onRemoveChannel(channelname)}
+          />
+        ))}
+      </Kb.Box2>
+    </Kb.Box2>
+  )
+}
+
+type ChannelInputProps = {onAdd: (channelname: string) => void; selected: Array<string>; teamID: Types.TeamID}
+
+const ChannelInputDesktop = ({onAdd, teamID}: ChannelInputProps) => {
+  const [filter, setFilter] = React.useState('')
+  const channels = Container.useSelector(s => Constants.getTeamChannelInfos(s, teamID))
+  const channelnames = [...channels.values()].map(c => `#${c.channelname}`)
+
+  const onSelect = value => {
+    onAdd(value)
+    setFilter('')
+  }
+
+  const {popup, popupAnchor, onKeyDown, setShowingPopup} = useAutocompleter(channelnames, onSelect, filter)
```

Comment:
perhaps we should only include channels in the autocompleter that aren't already chosen? it seems weird that `#general` is the first suggestion.